include/core/boundary_conditions.h

```diff
@@ -678,6 +681,162 @@ namespace BoundaryConditions
     prm.leave_subsection();
   }
 
+  /**
+   * @brief This class manages the boundary conditions for VOF solver
+   * It introduces the boundary functions and declares the boundary conditions
+   * coherently. (TODO update)
+   * The members ... contain double used for bc calculation :
+   *  - if bc type is "..." (... condition), "..." is the
+   * double passed to the deal.ii ConstantFunction
+   */
+
+  template <int dim>
+  class VOFBoundaryConditions : public BoundaryConditions<dim>
+  {
+  public:
+    std::vector<double> peeling_threshold;
+    std::vector<double> wetting_threshold;
+
+    void
+    declareDefaultEntry(ParameterHandler &prm, unsigned int i_bc);
+    void
+    declare_parameters(ParameterHandler &prm);
+    void
+    parse_boundary(ParameterHandler &prm, unsigned int i_bc);
+    void
+    parse_parameters(ParameterHandler &prm);
+  };
+
+  /**
+   * @brief Declares the default parameters for a boundary condition id i_bc
+   * i.e. Dirichlet condition (ConstantFunction) with value 0
+   *
+   * @param prm A parameter handler which is currently used to parse the simulation information
+   *
+   * @param i_bc The boundary condition id.
+   */
+  template <int dim>
+  void
+  VOFBoundaryConditions<dim>::declareDefaultEntry(ParameterHandler &prm,
+                                                  unsigned int      i_bc)
+  {
+    prm.declare_entry("type",
+                      "none",
+                      Patterns::Selection("none|peeling/wetting"),
+                      "Type of boundary condition for VOF"
+                      "Choices are <none|peeling/wetting>.");
+
+    prm.declare_entry("id",
+                      Utilities::int_to_string(i_bc, 2),
+                      Patterns::Integer(),
+                      "Mesh id for boundary conditions");
+
+    prm.declare_entry("peeling threshold",
+                      "0",
+                      Patterns::Double(),
+                      "Value (Double) for peeling threshold at bc");
+
+    prm.declare_entry("wetting threshold",
+                      "0",
+                      Patterns::Double(),
+                      "Value (Double) for wetting threshold at bc");
+  }
+
+  /**
+   * @brief Declare the boundary conditions default parameters
+   * Calls declareDefaultEntry method for each boundary (max 14 boundaries)
+   *
+   * @param prm A parameter handler which is currently used to parse the simulation information
+   */
+  template <int dim>
+  void
+  VOFBoundaryConditions<dim>::declare_parameters(ParameterHandler &prm)
+  {
+    this->max_size = 14;
```

Comment:
maybe rename this variable to max_bundary_size?

Reply:
I agree that it will be clearer, though it is defined in the same fashion than the `NSBoundaryConditions`, `HTBoundaryConditions` and `TracerBoundaryConditions`. Do you suggest that I change this parameter name everywhere?

Reply:
It's your decision, maybe we can do it in another PR. :)